framework/db/Command.php

```diff
@@ -1114,7 +1124,78 @@ protected function refreshTableSchema()
     }
 
     /**
-     * Resets [[sql]] and [[params]] properties.
+     * Marks the command to be executed in transaction.
+     * @param string|null $isolationLevel The isolation level to use for this transaction.
+     * See [[Transaction::begin()]] for details.
+     * @return $this this command instance.
+     * @since 2.0.14
+     */
+    protected function requireTransaction($isolationLevel = null)
+    {
+        $this->_isolationLevel = $isolationLevel;
+        return $this;
+    }
+
+    /**
+     * Sets a callable (e.g. anonymous function) that is called when [[Exception]] is thrown
+     * when executing the command. The signature of the callable should be:
+     *
+     * ```php
+     * function (\yii\db\Exception $e, $attempt)
```

Comment:
Would be good to describe `$e` and `$attempt`.

Reply:
Done.